### PR TITLE
Allow to override attibutes in devfile by factory URL parameters

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Allow to override attibutes in devfile by factory URL parameters.
Example query parameter:
`override.attributes.persistVolumes=true`

### What issues does this PR fix or reference?
#15500 

### Docs PR
https://github.com/eclipse/che-docs/pull/1031

### CODE REVIEW

File: wsmaster/che-core-api-workspace/src/main/resources/schema/1.0.0/devfile.json [hunk 1]
```diff
@@ -749,8 +749,7 @@
       },
       "additionalProperties": {
         "type": "string"
-      },
-      "javaType": "java.util.Map<String, String>"
+      }
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/main/resources/schema/1.0.0/devfile.json:752]: why this change ?

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java [hunk 2]
```diff
@@ -90,6 +91,10 @@ public JsonNode applyPropertiesOverride(
     return devfileNode;
   }
 
+  private String[] parseSegments(String key) {
+    return key.split("\\.", key.startsWith("attributes.") ? 2 : 0);
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java]: can you please write a comment here why it's doing what it's doing? I do understand it now, but I'm not sure in few months... It is definitely not a straight-forward and easy to read code.

Comment (sparkoo) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java]: or maybe use `key.replace("attributes.", "")` WDYT? Is it be more readable?

File: wsmaster/che-core-api-workspace/src/main/resources/schema/1.0.0/devfile.json (hunk 1 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-workspace/src/main/resources/schema/1.0.0/devfile.json:752]: just cleanup

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java (hunk 2 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/OverridePropertiesApplier.java]: added comment

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
crw-ci-test

**Comment by mshaposhnik** [comment]
ci-test

**Comment by mshaposhnik** [comment]
crw-ci-test

**Comment by sparkoo** [comment]
@mshaposhnik happy path tests are broken https://github.com/eclipse/che/pull/15618#pullrequestreview-340356886

**Review by sparkoo — APPROVED** [review]

**Comment by skabashnyuk** [comment]
CC @ibuziuk

**Review by skabashnyuk — APPROVED** [review]
